VideoStreaming/Networking/NetworkManager.swift

```diff
@@ -14,35 +14,68 @@ import Alamofire
 //    - https://api.twitch.tv/helix/streams?game_id=21779
 //    - https://api.twitch.tv/helix/videos?user_id=67955580
 
-protocol NetworkManager {
+enum APIError: Error {
+    case responseDataNil
+    case emptyDataArray
+    case jsonError(DecodingError)
+    case afError(Error)
+    case urlError(Error)
+    case unknownError(Error)
     
+    var localizedDescription: String {
+        switch self {
+            case .responseDataNil: return "Data is nil"
+            case .emptyDataArray: return "Data Array is empty"
+            case .jsonError(let jsonError): return jsonError.localizedDescription
+            case .afError(let afError): return afError.localizedDescription
+            case .urlError(let urlError): return urlError.localizedDescription
+            case .unknownError(let unknownError): return unknownError.localizedDescription
+        }
+    }
+}
+
+protocol NetworkManager {
     typealias QueryString = [String:Any]
-    
     var request: String { get }
-    
 }
 
 extension NetworkManager {
     static var headers: HTTPHeaders {
         return ["Client-ID": "xzpd1f4527fu8fct7p7own0pgi35v5"]
     }
     
-    func fetchData <T:Codable> (request: String, parameters: [String:Any] = [:], completion: @escaping (([T])->Void)) {
+    func fetchData <T:Codable> (request: String,
+                                parameters: QueryString = [:],
+                                completion: @escaping ((Bool, [T], APIError?)->Void)) {
```

Comment:
Perdón por llegar tarde a la conversación, :(. Pero yo creo que aquí aplicaría retornar un Result. Es algo nuevo del Swift 5, en donde devuelve un valor, o un error. Es que también veo que los argumentos de este completion son redundantes. Porque el primero Bool, y el arreglo de datos van muy ligados, si ven el primer parametro boleano es true si hay datos, y false para el resto, jeje. Con el result creo que quedaría más legible, y estaríamos aprovechando  un nuevo feature de Swift. https://www.hackingwithswift.com/articles/161/how-to-use-result-in-swift Ahí explican como usarlo. O que opina @castadev85 , ;)

Reply:
si, para mi también es redundante, lo puse por poner, si te parece lo saco

Reply:
lestoooo